### PR TITLE
Lab kinematic results - local processing

### PR DESCRIPTION
**Pull request by antoinefalisse**
This PR is meant to enable users to regenerate the paper results after downloading data from SimTK. The previous code was pulling data from the database, but the sessions aren't public so this was only possible for admins. This is also a good example of how to process data locally.

The dataset on SimTK should be slightly adjusted:
- The videos for computing the extrinsics are not there
- A couple other minor things

Everything is ready [here](https://drive.google.com/drive/folders/1VuhbZD3ZRFcGYkDU4IOFl8HC9t2kIleU?usp=sharing)
I adjusted the metadata to remove personal info. Maybe you want to take a quick look.
I will upload the datasets to SimTK when the PR is merged.

Also, the dataset on SimTK is structured differently as compared to what main expects, as part of labValidationVideosToKinematics, there is now a piece of code to re-structure the data. It basically copies over what is needed from the dataset to a different location and the processing is done from there. That means the newly processed data will not overwrite or be in the same folder as the dataset from SimTK, which is fine I guess.

I ran the code locally for one subject, and it worked fine. Let's ask Matt to test it (it can be done after merging).
Since I changed a couple of things in main and utilsDetector, I will do a proper testing of the app when merged in dev.

### CODE REVIEW

File: ReproducePaperResults/labValidationVideosToKinematics.py [hunk 1]
```diff
@@ -40,143 +43,225 @@
...
+# Since the prepint release, we updated a new augmenter model. To use the model
+# used for generating the paper results, select v0.1. To use the latest model
+# (now in production), select v0.2.
+augmenter_model = 'v0.2'
+
+# %% Data re-organization
+# To reprocess the data, we need to re-organize the data so that the folder
+# structure is the same one as the one expected by OpenCap. It is only done
+# once as long as the variable overwriteRestructuring is False. To overwrite
+# filpt the flag to True.
```
Comment (suhlrich) [inline ReproducePaperResults/labValidationVideosToKinematics.py:107]: flip the flag?

File: ReproducePaperResults/labValidationVideosToKinematics.py [hunk 2]
```diff
@@ -40,143 +43,225 @@
...
+                
+                # Detect if static trial with neutral pose to scale model.
+                if 'static' in trial.lower():                    
+                    scaleModel = True
+                else:
+                    scaleModel = False
+                
+                # Session specific intrinsic parameters
+                if 'subject2' in sessionName or 'subject3' in sessionName:
+                    intrinsicsFinalFolder = 'Deployed_720_240fps'
```
Comment (suhlrich) [inline ReproducePaperResults/labValidationVideosToKinematics.py:253]: these videos are 60fps. Am I forgetting something?

File: utilsDetector.py [hunk 3]
```diff
@@ -37,20 +37,25 @@ def runPoseDetector(CameraDirectories, trialRelativePath, pathPoseDetector,
         cameraDirectory = CameraDirectories_selectedCams[camName]
         print('Running {} for {}'.format(poseDetector, camName))
         if poseDetector == 'OpenPose':
-            runOpenPoseVideo(
+            extension = runOpenPoseVideo(
                 cameraDirectory,trialRelativePath,pathPoseDetector, trialName,
                 resolutionPoseDetection=resolutionPoseDetection,
                 generateVideo=generateVideo)
         elif poseDetector == 'mmpose':
             runMMposeVideo(
```
Comment (suhlrich) [inline utilsDetector.py:45]: seems we don't get the extension if we are using mmpose?

File: ReproducePaperResults/labValidationVideosToKinematics.py (hunk 2 above)
Comment (antoinefalisse) [reply ReproducePaperResults/labValidationVideosToKinematics.py:253]: The first two subjects were at 240fps

Comment (antoinefalisse) [reply ReproducePaperResults/labValidationVideosToKinematics.py:253]: First three actually
https://github.com/stanfordnmbl/opencap-core/blob/main/ReproducePaperResults/labValidationIDs.py#L14

File: utilsDetector.py (hunk 3 above)
Comment (antoinefalisse) [reply utilsDetector.py:50]: Fixed in latest commit

**Comment by antoinefalisse** [comment]
Thanks for the review - I addressed the comments.
I will make a proper testing on Ubuntu with HRNet later if you don't mind.
Will test dev pipeline with both openpose and hrnet when merged.

File: ReproducePaperResults/labValidationVideosToKinematics.py [hunk 4]
```diff
@@ -264,4 +264,6 @@ def process_trial(trial_name=None, session_name=None, isDocker=False,
                               resolutionPoseDetection=resolutionPoseDetection,
                               scaleModel=scaleModel, 
                               augmenter_model=augmenter_model,
-                              dataDir=dataDir)
\ No newline at end of file
+                              dataDir=dataDir)
+                
+            test=1
```
Comment (suhlrich) [inline ReproducePaperResults/labValidationVideosToKinematics.py]: remove

**Review by suhlrich — COMMENTED** [review]
looks good! ready for matt testing

**Pull request merged**